modules/library/cql/src/main/java/org/geotools/filter/text/commons/ExpressionToText.java

```diff
@@ -214,11 +220,45 @@ public Object visit(Multiply expression, Object extraData) {
 	@Override
 	public Object visit(PropertyName expression, Object extraData) {
 
-        StringBuilder output = asStringBuilder(extraData);        
-        output.append( expression.getPropertyName() );
+        StringBuilder output = asStringBuilder(extraData);
+        if(propertyNeedsDelimiters(expression)) {
+            output.append('"');
+            output.append( expression.getPropertyName().replace("\"", "\"\"") );
+            output.append('"');
```

Comment:
I can confirm what Kevin suggests, if you use single quotes that should be treated as a string, so if what you suggested works, that might be considered another bug (or an excess of leniency)